GetStream/stream-chat-android
# [4112] Improve video information design in the picker

### 🎯 Goal

closes #4112 

### 🛠 Implementation details

Transformed the video length information into a "chip".

### 🎨 UI Changes


#### Compose

| Before | After |
| --- | --- |
| ![Screenshot_1666371079](https://user-images.githubusercontent.com/37080097/197248559-58a327ff-5751-4e27-a3f3-cda2f99ce22e.png) | ![Screenshot_1666370926](https://user-images.githubusercontent.com/37080097/197248665-f9dfe999-c2e1-49ab-8699-f13b5482aef2.png) |

| Before | After |
| --- | --- |
| ![Screenshot_1666371069](https://user-images.githubusercontent.com/37080097/197248553-301c7871-2253-4465-a807-89093605183f.png) | ![Screenshot_1666370919](https://user-images.githubusercontent.com/37080097/197248661-3d1651da-991e-4d41-a588-d0e95f459be9.png) |

#### UI Components (XML)

| Before | After |
| --- | --- |
| ![Screenshot_1666371012](https://user-images.githubusercontent.com/37080097/197248821-4ee8335a-4dce-4314-9c43-7121d82c2db8.png) | ![Screenshot_1666370832](https://user-images.githubusercontent.com/37080097/197248885-9232e3d3-ab51-4647-acf0-cf57bed65132.png) |

| Before | After |
| --- | --- |
| ![Screenshot_1666371020](https://user-images.githubusercontent.com/37080097/197248828-4123301d-6bd4-4c44-8f58-f45873190bb9.png) | ![Screenshot_1666370860](https://user-images.githubusercontent.com/37080097/197248895-d8f20e04-f661-4429-b926-3d999f5cacd9.png) |

### 🧪 Testing

1. Make sure that you have videos stored on the phone/ emulator
2. Open the attachment picker on the media (image/video tab) and test out both light and dark theme

Expectation: All elements should be visible and readable

#### Customization test

**Tinting:**

<details>

<summary>Tinting Patch </summary>

```
Index: stream-chat-android-ui-components/src/main/res/values/styles.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-ui-components/src/main/res/values/styles.xml b/stream-chat-android-ui-components/src/main/res/values/styles.xml
--- a/stream-chat-android-ui-components/src/main/res/values/styles.xml	(revision d21654eec5718228831766a9f76f5dc61dd08efb)
+++ b/stream-chat-android-ui-components/src/main/res/values/styles.xml	(date 1666710912259)
@@ -1196,8 +1196,8 @@
         <item name="streamUiAttachmentsPickerVideoLengthTextSize">@dimen/stream_ui_text_small</item>
         <item name="streamUiAttachmentsPickerVideoLengthTextColor">@color/stream_ui_black</item>
         <item name="streamUiAttachmentsPickerVideoIconVisible">true</item>
-        <item name="streamUiAttachmentsPickerVideoIconDrawable">@drawable/stream_ui_ic_video</item>
-        <item name="streamUiAttachmentsPickerVideoIconDrawableTint">@color/stream_ui_black</item>
+        <item name="streamUiAttachmentsPickerVideoIconDrawable">@drawable/stream_ui_ic_reaction_thumbs_up</item>
+        <item name="streamUiAttachmentsPickerVideoIconDrawableTint">@color/stream_ui_accent_blue</item>
         <item name="streamUiAttachmentsPickerMediaAttachmentNoMediaText">@string/stream_ui_message_composer_no_files</item>
         <item name="streamUiAttachmentsPickerMediaAttachmentNoMediaTextSize">@dimen/stream_ui_text_large</item>
         <item name="streamUiAttachmentsPickerMediaAttachmentNoMediaTextColor">@color/stream_ui_text_color_primary</item>
```
</details>

1. Apply the patch above
2. Open the video picker

Expectation: A blue tinted thumbs up is now shown above videos

**Visibility:**

<details>

<summary>Visibility Patch </summary>

```
Index: stream-chat-android-ui-components/src/main/res/values/styles.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-ui-components/src/main/res/values/styles.xml b/stream-chat-android-ui-components/src/main/res/values/styles.xml
--- a/stream-chat-android-ui-components/src/main/res/values/styles.xml	(revision d21654eec5718228831766a9f76f5dc61dd08efb)
+++ b/stream-chat-android-ui-components/src/main/res/values/styles.xml	(date 1666711252301)
@@ -1192,7 +1192,7 @@
         <item name="streamUiAttachmentsPickerMediaAttachmentsTabIconDrawable">@drawable/stream_ui_attachment_permission_media</item>
         <item name="streamUiAttachmentsPickerAllowAccessToMediaButtonText">@string/stream_ui_message_composer_gallery_access</item>
         <item name="streamUiAttachmentsPickerAllowAccessToMediaIconDrawable">@drawable/stream_ui_attachment_permission_media</item>
-        <item name="streamUiAttachmentsPickerVideoLengthTextVisible">true</item>
+        <item name="streamUiAttachmentsPickerVideoLengthTextVisible">false</item>
         <item name="streamUiAttachmentsPickerVideoLengthTextSize">@dimen/stream_ui_text_small</item>
         <item name="streamUiAttachmentsPickerVideoLengthTextColor">@color/stream_ui_black</item>
         <item name="streamUiAttachmentsPickerVideoIconVisible">true</item>
```
</details>

1. Apply the patch above
2. Open the video picker

Expectation: The videos length `TextView` has its visibility set to gone. 
Try to also set `streamUiAttachmentsPickerVideoIconVisible` to false both with `streamUiAttachmentsPickerVideoLengthTextVisible` set to `false` and then it set to `true`. Setting both to false should remove the whole "card".

### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist

### 🎉 GIF

![gif](https://media4.giphy.com/media/xT1XGCYYoSewtnWf7i/giphy.gif?cid=ecf05e47xvsdz2wdb8fv28u3bulql7s7kll6z3lyvrp53puj&rid=giphy.gif&ct=g)
